static/app/views/automations/components/automationFormData.tsx

```diff
@@ -15,30 +16,45 @@ export function getNewAutomationData(
   data: AutomationFormData,
   state: AutomationBuilderState
 ): NewAutomation {
-  const stripDataConditionIds = (condition: any) => {
+  const stripDataConditionId = (condition: any) => {
     const {id: _id, ...conditionWithoutId} = condition;
+
+    if (condition.comparison?.filters) {
+      return {
+        ...conditionWithoutId,
+        comparison: {
+          ...condition.comparison,
+          filters: condition.comparison.filters?.map(stripSubfilterTypeAndId) || [],
```

Comment:
Why do we need to do this (strip the type and ID)? Could you add a code comment for future readers?

Reply:
good callout, added a comment above the function https://github.com/getsentry/sentry/blob/03a3aa28e6fe0158bdc52d4f3e298db9ace367b0/static/app/views/automations/components/automationFormData.tsx#L31